packages/react-native-executorch/common/rnexecutorch/models/ocr/DetectorUtils.cpp

```diff
@@ -0,0 +1,647 @@
+               float c, float centerThreshold) {
+  float smallestDistance = std::numeric_limits<float>::max();
+  std::size_t idx = -1;
+  float boxHeight = 0;
+  const Point centerOfCurrentBox = centerOfBox(currentBox);
+
+  for (std::size_t i = 0; i < boxes.size(); i++) {
+    if (ignoredIdxs.contains(i)) {
+      continue;
+    }
+    std::array<Point, 4> bbox = boxes[i].bbox;
+    const Point centerOfProcessedBox = centerOfBox(bbox);
+    const float distanceBetweenCenters =
+        distanceFromPoint(centerOfCurrentBox, centerOfProcessedBox);
+
+    if (distanceBetweenCenters >= smallestDistance) {
+      continue;
+    }
+
+    boxHeight = minSideLength(bbox);
+
+    const float lineDistance =
+        (isVertical
+             ? fabs(centerOfProcessedBox.x - (m * centerOfProcessedBox.y + c))
+             : fabs(centerOfProcessedBox.y - (m * centerOfProcessedBox.x + c)));
+
+    if (lineDistance < boxHeight * centerThreshold) {
+      idx = i;
+      smallestDistance = distanceBetweenCenters;
+    }
+  }
+
+  return idx != -1 ? std::optional(std::make_pair(idx, boxHeight))
+                   : std::nullopt;
+}
+
+std::vector<DetectorBBox>
+removeSmallBoxesFromArray(const std::vector<DetectorBBox> &boxes,
+                          float minSideThreshold, float maxSideThreshold) {
+  std::vector<DetectorBBox> filteredBoxes;
+
+  for (const auto &box : boxes) {
+    const float maxSide = maxSideLength(box.bbox);
+    const float minSide = minSideLength(box.bbox);
+    if (minSide > minSideThreshold && maxSide > maxSideThreshold) {
+      filteredBoxes.push_back(box);
+    }
+  }
+
+  return filteredBoxes;
+}
+
+static float minimumYFromBox(const std::array<Point, 4> &box) {
+  float minY = std::numeric_limits<float>::max();
+  for (const auto &pt : box) {
+    if (pt.y < minY) {
+      minY = pt.y;
+    }
+  }
+  return minY;
+}
+
+std::vector<DetectorBBox>
+groupTextBoxes(std::vector<DetectorBBox> &boxes, float centerThreshold,
+               float distanceThreshold, float heightThreshold,
+               int32_t minSideThreshold, int32_t maxSideThreshold,
+               int32_t maxWidth) {
+  // Sort boxes descending by maximum side length
+  std::sort(boxes.begin(), boxes.end(),
+            [](const DetectorBBox &lhs, const DetectorBBox &rhs) {
+              return maxSideLength(lhs.bbox) > maxSideLength(rhs.bbox);
+            });
+
+  std::vector<DetectorBBox> mergedVec;
+  float lineAngle;
+  while (boxes.size() > 0) {
+    DetectorBBox currentBox = boxes[0];
+    float normalizedAngle = normalizeAngle(currentBox.angle);
+    boxes.erase(boxes.begin());
+    std::unordered_set<std::size_t> ignoredIdxs;
+
+    while (true) {
+      // Find all aligned boxes and merge them until max_size is reached or no
+      // more boxes can be merged
+      auto [slope, intercept, isVertical] =
+          fitLineToShortestSides(currentBox.bbox);
+
+      lineAngle = atan(slope) * 180 / M_PI;
```

Comment:
And remove magic number
```suggestion
      lineAngle = std::atan(slope) * 180 / M_PI;
```